modules/generator/instance.go

```diff
@@ -553,13 +553,21 @@ func (i *instance) QueryRange(ctx context.Context, req *tempopb.QueryRangeReques
 			return nil, err
 		}
 	}
-
 	// Combine the raw results into the job results
 	walResults := rawEval.Results().ToProto(req)
 	jobEval.ObserveSeries(walResults)
 
 	r := jobEval.Results()
 	rr := r.ToProto(req)
+
+	maxSeries := int(req.MaxSeries)
+	if maxSeries > 0 && len(rr) > maxSeries {
+		return &tempopb.QueryRangeResponse{
+			Series: rr[:maxSeries],
+			Status: tempopb.PartialStatus_PARTIAL,
```

Comment:
Does this need to set message also?

Reply:
we can add it for consistency but this message is only relevant when returning to the user. The query-frontend combiner will check for the partial status and add it only at the very end. or do you think we should add it anyway?

Reply:
Ok, I misread and thought the message was already set in multiple places.  In that case a comment isn't really necessary either.

Reply:
this code is weird. it assumes there can be multiple local blocks processors, queries each and combines the results. is that possible?

Reply:
Yes it is part of the new rhythm architecture. There are 2 copies of local blocks, 1 is receiving grpc traffic and flushing to object storage (i.e. previous arch), the other is receiving queue traffic and not flushing (i.e. new arch). This is necessary to handle initial cutover and allows revert at any time. We can chat about this offline too.